### PR TITLE
inetaddr() more compliant with standards (especially error handling).

### PR DESCRIPTION
**Pull request by anpaza**
Standards-compliant inetaddr().

### CODE REVIEW

File: libs/libc/net/lib_inetaddr.c [hunk 1]
```diff
@@ -56,19 +56,65 @@
...
+      case 1:
+        {
+          result = a;
+          break;
+        }
+
+      case 2:
+        {
+          if (a < 0x100 && b < 0x1000000)
+            result = (a << 24) | b;
```
Comment (patacongo) [inline libs/libc/net/lib_inetaddr.c]: The coding standard requires that all if expresseions be enclosed in braces.

File: libs/libc/net/lib_inetaddr.c [hunk 2]
```diff
@@ -56,19 +56,65 @@
...
+          break;
+        }
+
+      case 3:
+        {
+          if (a < 0x100 && b < 0x100 && c < 0x10000)
+            result = (a << 24) | (b << 16) | c;
```
Comment (patacongo) [inline libs/libc/net/lib_inetaddr.c]: The coding standard requires that all if expresseions be enclosed in braces

File: libs/libc/net/lib_inetaddr.c [hunk 3]
```diff
@@ -56,19 +56,65 @@
...
+          break;
+        }
+
+      case 4:
+        {
+          if (a < 0x100 && b < 0x100 && c < 0x100 && d < 0x100)
+          result = (a << 24) | (b << 16) | (c << 8) | d;
```
Comment (patacongo) [inline libs/libc/net/lib_inetaddr.c]: The coding standard requires that all if expresseions be enclosed in braces

File: libs/libc/net/lib_inetaddr.c (hunk 1 above)
Comment (patacongo) [reply libs/libc/net/lib_inetaddr.c]: Let me be more clear.  You do not have to add any parentheses.  That was all fine.  You need braces.  This should be like:

    if (a < 0x100 && b < 0x1000000)
      {
        result = (a << 24) | b;
      }

Comment (patacongo) [reply libs/libc/net/lib_inetaddr.c]: the same is true with all of the other comments that mark as resolved but not fixed.

Comment (anpaza) [reply libs/libc/net/lib_inetaddr.c]: Sorry, I got you wrong.
Fixed.

**Comment by patacongo** [comment]
Thanks... I will merge the change once all of the checks complete.

**Comment by patacongo** [comment]
There were several successful build then the checks were cancelled.  I think that is good enough. Thanks for your patience.

File: libs/libc/net/lib_inetaddr.c [hunk 4]
```diff
@@ -56,19 +56,75 @@
...
+              result = (a << 24) | b;
+            }
+          break;
+        }
+
+      case 3:
+        {
+          if ((a < 0x100) && (b < 0x100) && (c < 0x10000))
+            {
+              result = (a << 24) | (b << 16) | c;
```
Comment (xiaoxiang781216) [inline libs/libc/net/lib_inetaddr.c]: @anpaza and @patacongo but should we call htos(c) here to get the network order? case 1/2 has the similar issue.

**Pull request merged**